### PR TITLE
[KOGITO-4913] - Serverless Workflows example: Order Processing with Knative

### PR DESCRIPTION
**Pull request by ricardozanini**
See: https://issues.redhat.com/browse/KOGITO-4913

This is a new example to demonstrate how to orchestrate events with Kogito Serverless Workflow Quarkus extension on Knative Eventing platform.

TODO:

- [x] Deploy on AWS OpenShift
- [x] Create resources for users to deploy the example on their own

Signed-off-by: Ricardo Zanini <zanini@redhat.com>

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

**WARNING! Please make sure you are opening your PR against `master` branch!**

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>
</details>

### CODE REVIEW

File: serverless-workflow-order-processing/README.md [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+6. On Kubernetes, build this example locally with the [Dockerfile](Dockerfile), then [push it](kogito-sw-openshift.yaml) to a third party registry.
+   For OpenShift, you can let the [cluster build it for you](kubernetes/kogito-sw-openshift.yaml)
+7. Expose the service, if on minikube [follow this tutorial](https://kubernetes.io/docs/tasks/access-application-cluster/ingress-minikube/). There's an `Ingress` already pre created in [kubernetes/ingress.yaml](kubernetes/ingress.yaml). [`NodePort` also works](https://kubernetes.io/docs/tasks/access-application-cluster/ingress-minikube/#deploy-a-hello-world-app).
+   On OpenShift, the operator will automatically expose the service for you.
+8. Run `curl` from the terminal like you did in the previously steps. 
+   To see what's going on, just query for one of the Knative service sinks created on step #5. 
+   You should see something like:
+
+```
+// replace with output
```
Comment (sutaakar) [inline serverless-workflow-order-processing/README.md]: missing output :)

File: serverless-workflow-order-processing/kubernetes/kogito-sw-openshift.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  #env:
+  # env can be used to set variables during build
+  #- name: MY_CUSTOM_ENV
+  #  value: "my value"
+  gitSource:
+    contextDir: serverless-workflow-order-processing
+    uri: 'https://github.com/kiegroup/kogito-examples'
+  # set your maven nexus repository to speed up the build time
+  #mavenMirrorURL:
+
```
Comment (sutaakar) [inline serverless-workflow-order-processing/kubernetes/kogito-sw-openshift.yaml]:
```suggestion
---
apiVersion: app.kiegroup.org/v1beta1
kind: KogitoRuntime
metadata:
  name: order-processing-workflow
spec:
  replicas: 1
  # we bind our application with Knative env
  infra:
    - kogito-knative-infra
```

File: serverless-workflow-order-processing/README.md (hunk 1 above)
Comment (sutaakar) [reply serverless-workflow-order-processing/README.md]:
```suggestion
☁️  cloudevents.Event
Validation: valid
Context Attributes,
  specversion: 1.0
  type: internationalShipping
  source: /process/shippinghandling
  id: f1193deb-9c4e-4e90-8aab-27fdba4ed740
  time: 2021-04-15T16:12:04.455905Z
Extensions,
  knativearrivaltime: 2021-04-15T16:12:04.657355879Z
  kogitoparentprociid: 836fbb3b-a68f-487b-9d96-cc93ca387ab9
  kogitoprocid: shippinghandling
  kogitoprocinstanceid: 1cc0d2b9-c5fb-445c-8229-5790e7a7b254
  kogitorootprocid: orderworkflow
  kogitorootprociid: 836fbb3b-a68f-487b-9d96-cc93ca387ab9
  kogitousertaskist: 1
Data,
  {"id":"f0643c68-609c-48aa-a820-5df423fa4fe0","country":"Brazil","total":500,"description":"iPhone 7","shipping":"international"}
```

**Review by sutaakar — APPROVED** [review]
Looks nice, works well.
Just see some suggestions above.

**Review by cristianonicolai — APPROVED** [review]

**Pull request merged**